etc/DependencyInstaller.sh

```diff
@@ -39,7 +39,21 @@ _installCentosPackages() {
         ruby-devel \
         tcl-devel
 
-    yum install -y https://www.klayout.org/downloads/CentOS_7/klayout-${klayoutVersion}-0.x86_64.rpm
+    if ! [ -x "$(command -v klayout)" ]; then
+      yum install -y https://www.klayout.org/downloads/CentOS_7/klayout-${klayoutVersion}-0.x86_64.rpm
+    else
+      currentver="$(klayout -v)"
+      currentver=$(echo $currentver | grep -o '[0-9]\+\.[0-9]\+\.[0-9]\+')
+      currentver=$(echo $currentver | tr -d '.')
+      requiredver=$(echo $klayoutVersion | tr -d '.')
+      if [ "$currentver" -ge "$requiredver" ]; then
+        echo "KLayout version greater than or equal to ${klayoutVersion}"
+      else
+        echo "KLayout version less than ${klayoutVersion}"
+        sudo yum remove -y klayout
+        yum install -y https://www.klayout.org/downloads/CentOS_7/klayout-${klayoutVersion}-0.x86_64.rpm
+      fi 
+    fi
```

Comment:
I think this should be refactored into a function that takes the URL as an argument so we can use it with other OSes -- then add it to the Debian/RHEL/Suse install functions. Also, replace for readability `currentver` -> `currentVersion` and `requiredver` `requiredVersion`

Reply:
@vvbandeira 
I checked with Ubuntu it automatically update or overwrite with KLayout installation. 
Only for Centos 7 currently it is exit from installer script.
```
klayout-0.28.8-0.x86_64.rpm                                                                                                                                    |  31 MB  00:00:03     
Examining /var/tmp/yum-root-yDnTtz/klayout-0.28.8-0.x86_64.rpm: klayout-0.28.8-0.x86_64
/var/tmp/yum-root-yDnTtz/klayout-0.28.8-0.x86_64.rpm: does not update installed package.
Error: Nothing to do
```

Reply:
Ok, then you can skip the refactoring. But please fix the readability issue above.

Also, there's a problem with your regex/logic. One example where it would fail:
- Version installed on user's system = 0.27.13
- Version set by klayoutVersion = 0.28.1
In this example, the currentver variable would be 02713 and the required version 0281. The if statement would day that the installed version is greater than the required version.

The above example is possible with the current versioning scheme used by klayout, see below:

![image](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/9001905/386e8657-3133-4db9-a8c9-46cbd8d476ba)

Reply:
@vvbandeira 
Thanks for catching the regex logic for >=. I will change equation.

Reply:
@vvbandeira 
Can check the regex logic now?
added `_versionCompare`